### PR TITLE
Git Q/A

### PR DESCRIPTION
**Pull request by 95nikhil**
Git Q/A

### CODE REVIEW

File: answer.txt [hunk 1]
```diff
@@ -1,4 +1,14 @@
 •What is a gitignore file and what should it include for a Java project?
+Ans. gitignore is used to skip or ingore the files that are not required to be commited like server configuration files specific to local, log files etc.For Java project  it usually 
+includes .class(compiled class files), .log(log files), .ctxt(BlueJ files), .mtj.tmp/(J2ME files), pacakage files(.jar,.zip) etc.
+
```
Comment (AdityaSai515) [inline answer.txt:4]: Please add a few more file types that we can add in .gitignore for java projects

File: answer.txt [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+includes .class(compiled class files), .log(log files), .ctxt(BlueJ files), .mtj.tmp/(J2ME files), pacakage files(.jar,.zip) etc.
+ 
 •What does fast-forward in Git mean?
+Ans. When the path bewtween two branches is not diverged from the linear path git pushes the pointer from source branch to the new branch. This is known as git fast-forward.
+
 •Why should you have different branches?
-•In a project how many remote Git repositories do you usually have?
\ No newline at end of file
+Ans. when large number of developers are working on same repository on different modules that can have changes in same files, branching in git allows each developer to create seperate
+branches and isolate them from main repository to work making the merge process easy without any conflicts.
```
Comment (AdityaSai515) [inline answer.txt:10]: Please inline the content in files properly

**Comment by AdityaSai515** [comment]
Please make the changes as necessary

**Comment by SER-515-mjf** [comment]
You can reuse the P/R to add more commits to fix your code based on the reviewer’s comment, then request the review again.

**Pull request closed without merging**